node/consensus/src/lib.rs

```diff
@@ -407,7 +407,7 @@ impl<N: Network> Consensus<N> {
             let block_latency = next_block_timestamp - current_block_timestamp;
 
             metrics::gauge(metrics::blocks::HEIGHT, next_block.height() as f64);
-            metrics::counter(metrics::blocks::TRANSACTIONS, next_block.transactions().len() as u64);
+            metrics::increment_gauge(metrics::blocks::TRANSACTIONS, next_block.transactions().len() as f64);
```

Comment:
No this will add `next_block.transactions().len()` to the existing value of the gauge. I think you should use the `metrics::gauge` method, which will set the current value of transactions per block.

Reply:
Since prometheus is a time series database, even if we set this gauge to be the existing number of transactions, it would not be per block. Since we scrape the endpoint every [15s ](https://github.com/AleoHQ/snarkOS/blob/testnet3/node/metrics/prometheus.yml#L2), I'm pretty confident that we would only get the reading of the current state, i.e. we would be "missing" data from one block.

I decided to make the metric increment because the metric name is `snarkos_blocks_transactions_total` and I figured this way we would capture all the data. Now, it is correct that this metric _should_ be a counter, since it is always increasing. However, our snarkVM implementation only exposes an `increment_counter` function that increases by the [constant 1](https://github.com/AleoHQ/snarkVM/blob/mainnet/metrics/src/lib.rs#L50-L52) .

I am of the opinion that this transaction metric should be a total. The best way to do this would be to change the increment counter function in snarkvm to take a value, rather than make it a constant, but switching the metric to a gauge was the lowest-lift method. Let me know what is preferred here.